bitshuffle.py

```diff
@@ -137,21 +143,48 @@ def main():
 
     args = parser.parse_args()
 
+    # Checks if no parameters were passed so the help can be printed
+    if len(sys.argv[1:]) == 0:
+        parser.print_help()
+        exit(0)
+
+    if args.version:
+        print("Version: bitshuffle v{0}".format(program_version))
+        exit(0)
+
     if args.filename is None:
         args.filename = os.path.basename(args.input)
 
-    if args.encode:
-        with open(args.input, 'rb') as f:
-            packets = encode_file(f, args.chunksize, args.compresslevel,
-                                  args.filename)
-            with open(args.output, 'w') as of:
-                for p in packets:
-                    of.write(p)
-                    of.write("\n\n")
+    # Encode & Decode smart inference
+    if not args.encode and not args.decode:
+        # Assume Decode
+        if args.output and args.input == '/dev/stdin':
+            if stdin.isatty():
+                # Infers decode
+                args.decode = True
 
-                of.flush()
+        # Assume Encode
+        if args.input and args.output == '/dev/stdout':
+            if not args.input == '/dev/stdin':
+                # Infers encode
+                args.encode = True
 
+    if args.encode:
+
+        if check_for_file(args.input):
+            with open(args.input, 'rb') as f:
+                packets = encode_file(f, args.chunksize, args.compresslevel,
+                                      args.filename)
+                with open(args.output, 'w') as of:
+                    for p in packets:
+                        of.write(p)
+                        of.write("\n\n")
+
+                    of.flush()
+        else:
+            quit('Error: Input file not found')
```

Comment:
Should exit with error code. See https://github.com/charlesdaniels/bitshuffle/issues/14